canonical/vanilla-framework
# Hero Pattern: Build Hero Pattern Implementation Documentation Page

## Done

- Builds new Hero implementation page
- Moves existing page to "Design guidelines"
- Some code cleanup of the hero macro to make it more presentable when its markup is rendered in a code snippet

Fixes [WD-13627](https://warthogs.atlassian.net/browse/WD-13627)

## QA

- Open [hero docs](https://vanilla-framework-5242.demos.haus/docs/patterns/hero) and verify that the needed examples are documented and explained adequately. 

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots
![image](https://github.com/user-attachments/assets/239943c4-327c-4b13-9ee3-a6aafc43ef09)

## Known issues

- There is some irregular indentation in the markup rendered in the example code snippets, due to macro usage. This is being tackled as a separate issue by @pastelcyborg in #5245.
- At a later date, we may want to make this page more navigable by adding multiple pages and allowing users to select which example of a given layout to view using [segmented control](https://vanillaframework.io/docs/patterns/segmented-control). This was decided against in the short term in order to avoid scope creep, as our existing documentation page system makes this require a rather high level of effort. See [MM thread](https://chat.canonical.com/canonical/pl/gr3ji35g4bn5udwf8rusatfrfo).


[WD-13627]: https://warthogs.atlassian.net/browse/WD-13627?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
